omp/multigrid/pgm_kernels.cpp

```diff
@@ -14,7 +14,6 @@
 
 #include <ginkgo/core/base/exception_helpers.hpp>
 #include <ginkgo/core/base/math.hpp>
-#include <ginkgo/core/multigrid/pgm.hpp>
```

Comment:
These includes should not be removed ?

Reply:
No, they *need* to be removed. The kernels don't depend on anything from the header, so it's not necessary. Besides, the pgm header now includes the mpi header through the distributed matrix, but those are not available, as MPI is not linked against the kernel libraries.
TBH, I'm a bit confused that it actually compiles in our tests. I suspect that the mpi headers are installed in some standard directory, and thus always available in our tests.

Reply:
the mpi header include is always wrapped with GINKGO_BUILD_MPI, so it will not be include if we do not build the mpi.
If we build the mpi, those are public of targets, and the tests are always linked to all gko library.

Reply:
I wasn't talking about compiling the tests specifically. Instead I'm meant our CI in general. So I'm confused that the kernels compile in our CI, since they are not linked against MPI.